pkg/flow/tracer_map.go

```diff
@@ -126,3 +126,23 @@ func (m *MapTracer) evictFlows(ctx context.Context, enableGC bool, forwardFlows
 	}
 	mtlog.Debugf("%d flows evicted", len(forwardingFlows))
 }
+
+func (m *MapTracer) aggregate(metrics []*ebpf.BpfFlowMetrics) *ebpf.BpfFlowMetrics {
+	if len(metrics) == 0 {
+		mtlog.Warn("invoked aggregate with no values")
+		return &ebpf.BpfFlowMetrics{}
+	}
+	aggr := &ebpf.BpfFlowMetrics{}
+	for _, mt := range metrics {
+		if mt != nil {
+			// eBPF hashmap values are not zeroed when the entry is removed. That causes that we
+			// might receive entries from previous collect-eviction timeslots.
+			// We need to check the flow time and discard old flows.
+			if mt.StartMonoTimeTs <= m.lastEvictionNs || mt.EndMonoTimeTs <= m.lastEvictionNs {
```

Comment:
```suggestion
			if mt.EndMonoTimeTs <= m.lastEvictionNs {
```

Can you please explain why we need to compare Start and End times here ?

It seems both curl on httpd page & ubuntu image download works fine when removing the compare on start :smile_cat:

Reply:
![image](https://github.com/netobserv/netobserv-ebpf-agent/assets/91894519/e86d221d-8516-4084-808f-2e2f14930eee)
![image](https://github.com/netobserv/netobserv-ebpf-agent/assets/91894519/58864543-a897-47c3-b1ad-9438e5f18b4f)

Reply:
> Can you please explain why we need to compare Start and End times here ?
> 
> It seems both curl on httpd page & ubuntu image download works fine when removing the compare on start 😸

that code was just a revert and that is what 1.3 has where u don't see the issue correct ?

Reply:
@msherif1234 could it be due to this code block being removed: (as said here https://github.com/netobserv/netobserv-ebpf-agent/pull/172#discussion_r1316971958  ) ?
```
        // it might happen that start_mono_time hasn't been set due to
        // the way percpu hashmap deal with concurrent map entries
        if (aggregate_flow->start_mono_time_ts == 0) {
            aggregate_flow->start_mono_time_ts = current_time;
        }
```

It seems like sometimes `start_mono_time_ts` is 0, which would indeed make the condition `mt.StartMonoTimeTs <= m.lastEvictionNs` be true ...

Reply:
yeah that is the proof that same flow won't always hit the same cpu core so we had to add this trick unfortuntely revert couldn't bring that code and I forget about it thanks for catching it now I tested both large and small curl and they are pretty accurate